pkg/distributor/forwarding/forwarding.go

```diff
@@ -234,6 +237,8 @@ func (f *forwarder) Forward(ctx context.Context, endpoint string, rules validati
 	}
 
 	go func() {
+		defer spanlog.Finish()
```

Comment:
nit: there's no early returns in this function so this call doesn't really need to be deferred, does it?